crypto/utils/utils.go

```diff
@@ -398,3 +397,12 @@ func ExtendHashOutput(salt, message []byte, outputByteLength int) []byte {
 	}
 	return result
 }
+
+func Pad(x []byte, n int) []byte {
+	pad := make([]byte, n-len(x))
```

Comment:
If `len(x)` is larger than `n`, this line will panic.

Reply:
Replace the implementation by:
if n-len(x) >= 0 {
	pad := make([]byte, n-len(x))
	return append(pad, x...)
}
return x[0:n]